deps/wazuh_testing/wazuh_testing/scripts/monitoring.py

```diff
@@ -0,0 +1,264 @@
+import csv
+import signal
+import sys
+import json
+import os
+import time
+import logging
+import subprocess
+import argparse
+import threading
+
+from wazuh_testing.api import make_api_call, get_api_details_dict
+
+
+metrics_monitoring_pid = None
+
+STOP_STATISTICS_MONITORING = False
+
+logger = logging.getLogger(__name__)
+
+
+def setup_logger(log_file, debug=False):
+    # Create a logger
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.DEBUG)
+
+    # Create a file handler and set level to DEBUG
+    logging_files = os.path.join(os.path.dirname(log_file), "monitoring_script_logging.log")
+    file_handler = logging.FileHandler(logging_files)
+
+    if debug:
+        file_handler.setLevel(logging.DEBUG)
+    else:
+        file_handler.setLevel(logging.INFO)
+
+    # Create a formatter and set it to the handler
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    file_handler.setFormatter(formatter)
+
+    # Add the file handler to the logger
+    logger.addHandler(file_handler)
+
+    return logger
+
+
+def signal_handler(sig, frame):
+    global metrics_monitoring_pid
+
+    logger.info("Signal received. Exiting...")
+
+    if metrics_monitoring_pid:
+        os.kill(metrics_monitoring_pid, signal.SIGKILL)
+
+    sys.exit(0)
+
+
+def create_csv_header(process, directory):
+    file_path = os.path.join(directory, f'{process}_stats.csv')
+
+    with open(file_path, 'w', newline='') as file:
+        writer = csv.writer(file)
+        if process == "wazuh-db":
+            writer.writerow(["timestamp", "name", "queries_received", "queries_global", "queries_wazuhdb",
+                             "time_execution", "time_global", "time_wazuhdb"])
+        elif process == "wazuh-remoted":
+            writer.writerow(["timestamp", "name", "bytes_received", "bytes_sent", "keys_reload_count",
+                             "received_keepalive",
+                             "received_request", "received_shutdown", "received_startup", "received_discarded",
+                             "received_event", "sent_ack", "sent_ar", "sent_discarded", "sent_request", "sent_sca",
+                             "sent_shared", "queue_size", "queue_usage"])
+        elif process == "wazuh-analysisd":
+            writer.writerow(["timestamp", "name", "bytes_received", "processed_events", "processed_events_received",
+                             "decoded_agent", "Decoded Events Syscheck", "dropped_agent", "dropped_syscheck",
+                             "writte_breakdown_alerts", "queue_size_alerts", "queue_usage_alerts",
+                             "queue_syscheck_size", "queue_syscheck_usage"])
```

Comment:
```suggestion
            writer.writerow(["timestamp", "name", "bytes_received", "processed_events", "processed_events_received",
                             "decoded_agent", "decoded_syscheck_events", "dropped_agent", "dropped_syscheck",
                             "writte_breakdown_alerts", "queue_size_alerts", "queue_usage_alerts",
                             "queue_syscheck_size", "queue_syscheck_usage"])
```

Reply:
Done in 6a5a9a0b08b16df2ba3fb3ad606b875be998b05d